drivers/gpio/gpio_gecko.c

```diff
@@ -91,18 +93,31 @@ static int gpio_gecko_configure(struct device *dev,
 		return -ENOTSUP;
 	}
 
-	if ((flags & GPIO_DIR_MASK) == GPIO_DIR_IN) {
-		if ((flags & GPIO_PUD_MASK) == GPIO_PUD_PULL_UP) {
+	if (flags & GPIO_OUTPUT) {
+		/* Following modes enable both output and input */
+		if (flags & GPIO_SINGLE_ENDED) {
+			if (flags & GPIO_LINE_OPEN_DRAIN) {
+				mode = gpioModeWiredAnd;
+			} else {
+				mode = gpioModeWiredOr;
+			}
+		} else {
+			mode = gpioModePushPull;
+		}
+		out = (flags & GPIO_OUTPUT_INIT_HIGH) ? 1U : 0U;
+	} else if (flags & GPIO_INPUT) {
+		if (flags & GPIO_PULL_UP) {
 			mode = gpioModeInputPull;
 			out = 1U; /* pull-up*/
-		} else if ((flags & GPIO_PUD_MASK) == GPIO_PUD_PULL_DOWN) {
+		} else if (flags & GPIO_PULL_DOWN) {
```

Comment:
Similar comment here. I think pull up and pull down both set should result in -EINVAL. Is there a good reason not to?

In general, if one flag is meant to take precedence over another, I think some static inline helpers like `gpio_flags_pull_up()`, `gpio_flags_pull_down()`, etc. should be added which respect the precedence ordering, which can be called from all drivers. This will ensure consistent behavior when moving programs between SoCs. (But I still personally think -EINVAL is a little better.)